bundles/org.openhab.voice.googlestt/src/main/java/org/openhab/voice/googlestt/internal/GoogleSTTService.java

```diff
@@ -272,16 +274,18 @@ && isExpiredInterval(maxSilenceMillis, responseObserver.getLastInputTime())) {
                 logger.debug("Stops listening, max silence time reached");
                 break;
             }
-            if (dataN != readBytes) {
+            if (numBytesRead != remaining) {
+                remaining = remaining - numBytesRead;
```

Comment:
It believe this will be wrong when `numBytesRead` is -1 because we reached the end of the stream?

Reply:
I didn't think about this as those services are only used with real time audio right now, but it's a good catch, better have the Stt services ready to support a finite stream.